File: SteamFitter.java
/* SteamFitter
* @Bryspeelm
* Description:
* Goal: Identify extra pipes and what line they occur in a .tsv file.
* Ver 1.0
*/

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
public class SteamFitter{

    public static void main(String[] args) {
        int numOfPipes = 23; // !!do not change unless number fields change!!
        int lineNum = 1; // start at one to take header into account
        String filename = "community_users.tsv";

        // Find pipes out of range in community_users.tsv file
        try{
            //Open file
            BufferedReader reader = new BufferedReader(new FileReader(filename));
            String line;
            int count;

            //Read lines and count number of pipes
            while ((line = reader.readLine()) != null)
            {
                count =0; // reset pipe counter for each new line
                //iterate through line and find pipes
                char[] chars = line.toCharArray();
                for (char ch : chars) {
                   if(ch == '|'){
                       count++;
                   }
                }

                //print line that has too many pipes along with the number of pipes
                if (count > numOfPipes){
                    System.out.println((count - numOfPipes) + " extra pipes @ line " +
                            lineNum);
                }
                lineNum++;
            }
        }
        catch(IOException ex){
            System.out.println("Cannot find file");
        }
    }
}

File: SteamFitter.py
# SteamFitter
# @Bryspeelm
# Description: 
# Goal: Identify extra pipes and what line they occur in a .tsv file.
# Ver 1.0

# Main Method
def main():
    numOfPipes = 23 # !!do not change unless number fields change!!
    lineNum = 1 # start at one to take header into account

#Find pipes out of range in community_users.tsv file
    try:
        #Open file
        infile = open('community_users.tsv', 'r')
        #read lines and count number of pipes
        for line in infile:
            count = 0 # reset pipe counter for each new line
            for char in line:
                if char == '|':
                    count+=1
            #print line that has too many pipes along with the number of pipes
            if count > numOfPipes:
                print(count - numOfPipes), " extra pipes @ line ", lineNum 
                within(line) # call within function
            lineNum+=1 # move line counter to next line
    except IOError:
        print("Cannot find file")

# find index of pipe inside quotes
def within(ln):
    charNum = 0; #var that keeps track of char place
    flag =0;
    qtCt = 0; # track quote count
    #Look for quotes in line and switch flag on/off
    for char in ln:
        charNum+=1; #incriment char place
        # Alternate Flag
        if char == '\"':
            qtCt+=1
            if flag == 0:
                flag = 1
            else: 
                flag = 0
        if char == '|' and flag == 1 and qtCt % 2 != 0:
            print("Pipe in quotes at char #: ", charNum)
#call main function
main()